hathor/transaction/poa/poa_block.py

```diff
@@ -0,0 +1,78 @@
+#  Copyright 2024 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from hathor.consensus import poa
+from hathor.transaction import Block, TxVersion
+from hathor.transaction.storage import TransactionStorage
+from hathor.transaction.util import VerboseCallback, int_to_bytes, unpack, unpack_len
+
+# Size limit in bytes for signature field
+_MAX_POA_SIGNATURE_LEN: int = 100
+
+
+class PoaBlock(Block):
+    """A Proof-of-Authority block."""
+
+    def __init__(
+        self,
+        timestamp: int | None = None,
+        signal_bits: int = 0,
+        weight: float = 0,
+        parents: list[bytes] | None = None,
+        hash: bytes | None = None,
+        data: bytes = b'',
+        storage: TransactionStorage | None = None,
+        signer_id: bytes = b'',
+        signature: bytes = b'',
+    ) -> None:
+        super().__init__(
+            nonce=0,
+            timestamp=timestamp,
+            signal_bits=signal_bits,
+            version=TxVersion.POA_BLOCK,
+            weight=weight,
```

Comment:
Should we remove weight?

Reply:
The `weight` is used to keep compatibility with the current `BlockConsensusAlgorithm`. It is used to signal whether a block is in turn or out of turn.

Reply:
Got it. I guess it's ok to move forward like this. But shouldn't we write a new PoA Consensus Algorithm?

Reply:
Thanks!